### PR TITLE
update fortran model stub code

### PR DESCRIPTION
**Pull request by mattldawson**
adds fortran version of `my_model_t` and `my_state_t` (untested)

### CODE REVIEW

File: src/model/cpp_model.F90 [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ! SPDX-License-Identifier: Apache-2.0
 !
 !> \file
-!> Fortran wrapper for aerosol models in C++
```
Comment (jeff-cohere) [inline src/model/cpp_model.F90:5]: I think these boilerplate comments are more trouble than they're worth. In most cases in the wild, they're out of date.

File: src/model/cpp_model.F90 [hunk 2]
```diff
@@ -53,18 +56,18 @@ end function aero_bridge_cpp_new_model
 
   !> Creates a C++ aerosol model
   !!
-  !! If the model is not a supported model in C++, a null pointer is returned
+  !! If the model is not a supported model in C, a null pointer is returned
```
Comment (jeff-cohere) [inline src/model/cpp_model.F90]: Isn't this a C++ model?

File: src/state/state.F90 [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+!> The state_t type and related functions
+module aero_state
+
+  implicit none
+  private
+
+  public :: state_t
+
+  type, abstract :: state_t
```
Comment (jeff-cohere) [inline src/state/state.F90:15]: Do we need anything to add a finalizer here? Or if you do it on an extending class, does it Just Work(tm)?

**Review by jeff-cohere — APPROVED** [review]
Looks great. This reminds me that I have to go back and make sure I compute the optics from the products of the different quantities in the C/C++ models. But one way to force that issue is to have a test for all models and make sure they get the right answers.

File: src/state/state.F90 (hunk 3 above)
Comment (mattldawson) [reply src/state/state.F90:15]: yeah, extending classes can add finalizers even if the base class doesn't have one (and I'm not sure if it's even possible to add a finalizer to an abstract class - it's only argument is a non-polymorphic variable)

File: src/model/cpp_model.F90 (hunk 2 above)
Comment (mattldawson) [reply src/model/cpp_model.F90]: copy/paste error =) I didn't do anything with the c/c++ model fortran code other than just get it to compile

**Comment by mattldawson** [comment]
thanks! that sounds like a good idea - maybe I'll start on a couple tests for the fortran "my model" code

File: src/model/cpp_model.F90 (hunk 1 above)
Comment (mattldawson) [reply src/model/cpp_model.F90:5]: I'm ok with removing them. I'll add this to #29

**Pull request merged**